src/schema/sources.ts

```diff
@@ -808,6 +825,29 @@ export const resolvers: IResolvers<any, Context> = {
         },
       );
     },
+    checkUserMembership: async (
+      _,
+      { sourceId, memberId }: CheckUserMembershipArgs,
+      ctx,
+      info,
+    ): Promise<GQLSourceMember> => {
+      await ensureSourcePermissions(ctx, sourceId, SourcePermissions.View, {
+        primaryMemberId: memberId,
+      });
```

Comment:
Why do we check this?
Technically a blocked user would have no rights to view.

My initial idea was we could simply query the user and return "false" or "true" based on their role?

Reply:
Ah yes, but this seems to abuse the function.
For the specced item now a query should be more than enough no?

Reply:
Good point. Yes, a query should be enough, just thought of making one endpoint to fetch source membership as we don't have one.

Reply:
Agree with Chris. This is very confusing, we need to verify that the logged user has access to make this check and a simple query should be fine to check the blocked user